File: docs/snippets/csharp/metadata/PageBoundaries.cs
using Kreuzberg;
using System.Text;

var result = Kreuzberg.ExtractFileSync("document.pdf");

if (result.Metadata.Pages?.Boundaries != null)
{
    var contentBytes = Encoding.UTF8.GetBytes(result.Content);

    foreach (var boundary in result.Metadata.Pages.Boundaries.Take(3))
    {
        var pageBytes = contentBytes[boundary.ByteStart..boundary.ByteEnd];
        var pageText = Encoding.UTF8.GetString(pageBytes);

        Console.WriteLine($"Page {boundary.PageNumber}:");
        Console.WriteLine($"  Byte range: {boundary.ByteStart}-{boundary.ByteEnd}");
        Console.WriteLine($"  Preview: {pageText[..100]}...");
    }
}

File: docs/snippets/php/metadata/page_boundaries.php
```php
<?php

declare(strict_types=1);

/**
 * Page Boundary Tracking
 *
 * Access page boundary information to extract content from specific pages
 * using byte offsets in the extracted content.
 */

require_once __DIR__ . '/vendor/autoload.php';

use function Kreuzberg\extract_file;

$result = extract_file('document.pdf');

// Check if page boundaries are available
if (isset($result->metadata->pages->boundaries) && !empty($result->metadata->pages->boundaries)) {
    $boundaries = $result->metadata->pages->boundaries;
    $contentBytes = $result->content;

    // Process first 3 pages
    $pagesToShow = array_slice($boundaries, 0, 3);

    foreach ($pagesToShow as $boundary) {
        // Extract content for this page using byte offsets
        $pageContent = mb_substr(
            $contentBytes,
            $boundary->byteStart,
            $boundary->byteEnd - $boundary->byteStart
        );

        echo "Page {$boundary->pageNumber}:\n";
        echo "  Byte range: {$boundary->byteStart}-{$boundary->byteEnd}\n";
        echo "  Preview: " . mb_substr($pageContent, 0, 100) . "...\n\n";
    }
}
```
